src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -0,0 +1,478 @@
+        this.setState({ snackbarOpen: true });
+      } else if (applicants.some((applicant) => applicant.AD_Username === username)) {
+        // Display an error if the selected user is already in the list
+        this.snackbarText = String(applicantProfile.fullName) + ' is already in the list.';
+        this.snackbarSeverity = 'info';
+        this.setState({ snackbarOpen: true });
+      } else {
+        // Add the profile object to the list of applicants
+        applicants.push(applicantProfile);
+        this.setState({ applicants });
+        if (this.state.applicants.some((applicant) => applicant.AD_Username === username)) {
+          this.snackbarText =
+            String(applicantProfile.fullName) + ' was successfully added to the list.';
+          this.snackbarSeverity = 'success';
+          this.setState({ snackbarOpen: true });
+        }
+      }
+    } catch (error) {
+      this.snackbarText = 'Something went wrong while trying to add this person. Please try again.';
+      this.snackbarSeverity = 'error';
+      this.setState({ snackbarOpen: true });
+    }
+  }
+
+  /**
+   * Callback for changing the primary applicant
+   * @param {String} profile The StudentProfileInfo object for the person who is to be made the primary applicant
+   */
+  handleChangePrimary = (profile) => {
+    this.setState({ updating: true });
+    if (profile) {
+      if (this.state.applicants.includes(profile)) {
+        this.setState({ newPrimaryApplicant: profile, editDialogOpen: true });
+      }
+    }
+  };
+
+  handleChangePrimaryAccepted = () => {
+    if (this.state.newPrimaryApplicant) {
+      try {
+        this.saveApplication(this.state.newPrimaryApplicant.AD_Username, this.state.applicants);
+      } catch (error) {
+        this.snackbarText = 'Something went wrong while trying to save the new primary applicant.';
+        this.snackbarSeverity = 'error';
+        this.setState({ snackbarOpen: true, saving: 'failed' });
+      }
+      this.handleCloseOkay();
+    }
+  };
+
+  /**
+   * Callback for applicant list remove button
+   * @param {String} profileToRemove Username for student
+   */
+  handleRemove = (profileToRemove) => {
+    this.setState({ updating: true });
+    if (profileToRemove) {
+      let applicants = this.state.applicants; // make a separate copy of the array
+      let index = applicants.indexOf(profileToRemove);
+      if (index !== -1) {
+        applicants.splice(index, 1);
+        this.setState({ applicants });
+      }
+    }
+    this.setState({ updating: false });
+  };
+
+  /**
+   * Callback for apartment application save button
+   */
+  handleSaveButtonClick = () => {
+    let debugMessage = 'DEBUG: Save button was clicked'; //! DEBUG
+    console.log(debugMessage); //! DEBUG
+    // The method is separated from callback because the housing API service must be handled inside an async method
+    this.saveApplication(this.state.userProfile.AD_Username, this.state.applicants);
```

Comment:
Change this.state.userProfile.AD_Username to this.state.primaryUsername

Reply:
The request change has been made and committed